index.js

```diff
@@ -1,18 +1,54 @@
 import React from 'react';
-import {AppRegistry, StyleSheet, Text, View} from 'react-native';
+import {AppRegistry, StyleSheet, Text, View, Button, NativeModules} from 'react-native';
 
 class RNHighScores extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      scores: props.scores
+    };
+
+    this._exitButtonPressed = this._exitButtonPressed.bind(this);
+    this._clearButtonPressed = this._clearButtonPressed.bind(this);
+  }
+
+  _exitButtonPressed() {
+    NativeModules.F3HBridgeModule.exitHighScoreButtonTapped(this.state.scores)
+  }
+
+  _clearButtonPressed() {
+    this.setState({
+      scores: []
+    })
+  }
+
   render() {
-    var contents = this.props['scores'].map((score) => (
-      <Text key={score.name}>
-        {score.name}:{score.value}
-        {'\n'}
+    var scores = this.state.scores;
+
+    var scoreContent = scores.map((score, i) => (
+      <Text key={'score-' + i}>
+        {score.value} - {score.createdAt}
+        {"\n"}
       </Text>
-    ));
+    ))
```

Comment:
A set of text labels like this will fit on the screen, but if you collected more scores you'd eventually want a scrolling list. (Is this why you limit it to top 3?)

I think the scores should be displayed using a FlatList, or possibly a SectionList, partly because that is more flexible and partly because the point of this whole exercise is about learning React Native, and so if we avoid using the more complicated components, we're just depriving ourselves of learning how they work.

Reply:
I switched to FlatList

Reply:
I compiled and ran the app:

* When there are no scores, the "No scores yet!" message is flush with the left side of the screen, no margin. It should be centered.
* I generated some high scores and managed to create two with the same timestamp, which resulted in a "two children with the same key" warning on this view.
* I wasn't sure what to expect, but FlatList is rendered as a scrollable list between the stationary Text component above and the Button components below. So we kinda sorta get the effect of a title/nav bar and a button bar.
* The Text title is displayed under the status bar; on iOS this is handled by UINavigationController; I'm not sure what the cross-platform way to do this would be.